### PR TITLE
Various improvements ( tx page, sync, load ) & codebase cleanup

### PR DESCRIPTION
**Pull request by btenterprise2020**
_Ready to merge_

### Breaking Changes

+ Modification on DB, need resync to reflect information
+ Upgraded to web3 1.0, node connection changed to WebSocket ( please open WebSocket connection on your node side)
+ Requires https://coinmarketcap.com/api/ account & api_key value to sync price info

### Changelog

- [x] Upgraded to web3 1.0, now supports more tx information with faster sync & load speed! ( About 10x faster than http rpc )
- [x] Fetch ERC20 tokens & Contracts from network, record them on DB
- [x] eslint & html-linter added to improve collaborative environment for decentralized buiild.
- [x] Improve web3relay by fetching tx information from DB
- [x] Add FIAT exchange for addresses and tx
- [x] Drop dao / bomb-chart support for lower priority
- [x] Add documentation about mongodb authentication
- [x] Enforce lowercase on transactions & accounts
- [x] Bug fixes & small improvements

### Preview

Capture

Capture

### Demo page

https://classic.btenterprise.io/home

### Donation

If you would like to support my work here is my tip jar.

ETC: 0xbfb05c94cde31adcc85636a5faa15d6e8f03478e

QR

tested on local environment

cc @realcodywburns @pyskell @hackmod

### CODE REVIEW

**Comment by btenterprise2020** [comment]
ps) since web3 1.0 lacks support for `web3.createbatch` related documentation, I've excluded batch job support from `tools/sync.js` and `tools/richlist.js` ( still faster than before thanks to ws ❤️) @hackmod

**Comment by btenterprise2020** [comment]
current sync speed using updated `tools/sync.js` : 100 blocks per second 🎉

**Comment by btenterprise2020** [comment]
Restored Geth / Geth-Classic support but not tested with them ( Geth & Parity supports Websocket however does geth-classic supports it also? )

**Comment by btenterprise2020** [comment]
PR finalized for now, will commit more on next PR!

File: tools/sync.js [hunk 1]
```diff
@@ -9,63 +9,88 @@ var etherUnits = require("../lib/etherUnits.js");
...
-var async = require('async');
+var asyncL = require('async');
 var Web3 = require('web3');
 
 var mongoose        = require( 'mongoose' );
 var Block           = mongoose.model( 'Block' );
 var Transaction     = mongoose.model( 'Transaction' );
 var Account         = mongoose.model( 'Account' );
 
+const normalizeTX = async (txData, blockData) => {
```
Comment (phyro) [inline tools/sync.js]: this function seems to be defined the same in 2 different files, should we abstract it out?

**Comment by phyro** [comment]
I advise you open a smaller PR next time, it's quite hard to review such big PRs. I've not tested it locally, I briefly skimmed over the code for now and did not find a malicious attempt (should be checked also by others as well). Thanks for the docker-compose

File: tools/sync.js (hunk 1 above)
Comment (btenterprise2020) [reply tools/sync.js]: I think so :)

**Comment by btenterprise2020** [comment]
Wasn't possible to open with smaller PRs, merging conflicts and the pr is actually a big cleanup so :) ( Will be same if we have eslint for PR )

**Comment by btenterprise2020** [comment]
eth.contract related bug fixed :)

**Comment by btenterprise2020** [comment]
@phyro I believe this PR is finalized and ready to go 🎉

**Comment by hackmod** [comment]
web3.js 1.0 is not matured. It is beta state. we don't have to hurry to use web3 1.0 at all
https://github.com/ethereum/web3.js/releases

**Comment by btenterprise2020** [comment]
@hackmod In that perspective Web3 0.20.x version is deprecated and unsecure https://medium.com/@samuel_91690/web3-js-v1-0-0-beta-38-8d2bb6e73d0b , stealing the web3 maintainer's word: `The version 0.20.x is in maintenance mode since 19.04.2018 and just got security updates and smaller bug fixes. Because of the coming stable release of the library, I’ve deprecated version 0.20.x and recommend switching to version 1.0.0-x of the library if you have not already. Further details of the roadmap of the library are explained below.`

**Comment by btenterprise2020** [comment]
I've used web3 1.0 cause they are being used on ganache-cli , mist, and metamask. If web3 1.0 is not matured, why are they using it?

File: public/tpl/header.html [hunk 2]
```diff
@@ -44,7 +44,7 @@
                 <li class="active">
                     <a href="/">Home</a>
                 </li>
-                <li ng-show="settings.showRichList">
```
Comment (hackmod) [inline public/tpl/header.html]: why remove this line without proper explanation?

one can disable `settings.showRichList = false` while `$ node tools/richlist.js` sync or any richlist maintaining step.

Comment (btenterprise2020) [reply public/tpl/header.html]: Is there any reason to disable them while it is being synced by default? ( using `tools/sync.js` )

Comment (btenterprise2020) [reply public/tpl/header.html]: Addressed your concern @hackmod , thanks for reviewing.

**Comment by btenterprise2020** [comment]
Demo page is now live https://classic.btenterprise.io.

Anyone still not willing to review or merge? What am I contributing for? 😂

File: tools/sync.js [hunk 3]
```diff
@@ -75,21 +75,20 @@ const normalizeTX = async (txData, receipt, blockData) => {
...
+  if (receipt.status) {
+    tx.status = receipt.status;
+  }
 
   if (txData.to) {
     tx.to = txData.to.toLowerCase();
     return tx;
-  }
-  if (tx.creates) {
+  } else if (tx.creates) {
```
Comment (phyro) [inline tools/sync.js]: @btenterprise2020 can this ever be true? the tx is defined above and it doesn't have `creates` defined, am I missing something?

Comment (btenterprise2020) [reply tools/sync.js]: Good catch, it should be txData.creates field that is supported by Parity.

**Comment by hackmod** [comment]
etc explorer use parity heavily. we have to make new dev branch like as web3-1.0 or something
it causes break all parity behaviors.

**Comment by btenterprise2020** [comment]
@hackmod What do you mean? I am using Parity.

**Comment by hackmod** [comment]
ok. I see now. you've extract parity extension stuff to the web3-explorer.
https://github.com/btenterprise2020/web3-explorer.js/blob/master/index.js

why are you trash all batch job stuff with no reason?

**Comment by btenterprise2020** [comment]
Batch job is no longer necessary for web3 1.0, if you want them please make a pr.

**Comment by hackmod** [comment]
and do not rebrand original npm package https://github.com/axic/web3-complete.js

**Comment by btenterprise2020** [comment]
@hackmod If you have any more further discussions, just let me know at DM. We have no reason to spam here. Thanks.

**Pull request merged**